### PR TITLE
feat: add ahp weights into TOPSIS Method

### PR DESCRIPTION
**Pull request by storyofhis**
- add ahp weights into TOPSIS method

### CODE REVIEW

File: src/app/siswa/rekomendasi/_components/topsis-card.tsx [hunk 1]
```diff
@@ -0,0 +1,68 @@
+"use client";
+
+import { Card, CardContent, CardDescription, CardHeader, CardTitle } from "@/components/ui/card";
+import { Tabs, TabsContent, TabsList, TabsTrigger } from "@/components/ui/tabs";
+import { Role, TopsisWeight } from "@/types/data/common";
+import { RecommendationResult } from "@/types/data/recommendation";
```
Comment (albugowy15) [inline src/app/siswa/rekomendasi/_components/topsis-card.tsx:6]: Prefer type import. Contoh:
```ts
import {type Role, type TopsisWeigth} from "@/types/data/common"
```

File: src/app/siswa/rekomendasi/_components/topsis-card.tsx [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+import { zodResolver } from "@hookform/resolvers/zod";
```
Comment (albugowy15) [inline src/app/siswa/rekomendasi/_components/topsis-card.tsx:7]: Import ini gak kepake, hapus aja

File: src/app/siswa/rekomendasi/_components/topsis-card.tsx [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+
+interface TopsisCardProps {
+    data: RecommendationResult[];
+    topsis_ahp: RecommendationResult[];
+}
+
+  
+export const TopsisCard: React.FC<TopsisCardProps> = ({data, topsis_ahp}) => {
+    const [tab, setTab] = useState<TopsisWeight>("entropy");
+    const [buttonLoading, setButtonLoading] = useState(false);
```
Comment (albugowy15) [inline src/app/siswa/rekomendasi/_components/topsis-card.tsx:19]: Ini gak kepake, hapus aja

File: src/app/siswa/rekomendasi/_components/topsis-card.tsx (hunk 1 above)
Comment (albugowy15) [reply src/app/siswa/rekomendasi/_components/topsis-card.tsx:6]: Yang import role gak kepake, hapus

File: src/lib/api.ts [hunk 4]
```diff
@@ -46,6 +46,8 @@ export async function protectedFetch<T>(
     });
 
     const json = (await res.json()) as APIResponse<T>;
+    /* eslint-disable */
+    // console.log("topsis ahp", json.data?.topsis_ahp)
```
Comment (albugowy15) [inline src/lib/api.ts:50]: hapus

**Review by albugowy15 — CHANGES_REQUESTED** [review]
Jangan lupa run command `pnpm format` sebelum push ke repo

File: src/app/siswa/rekomendasi/_components/topsis-card.tsx (hunk 1 above)
Comment (storyofhis) [reply src/app/siswa/rekomendasi/_components/topsis-card.tsx]: done!!

File: src/app/siswa/rekomendasi/_components/topsis-card.tsx (hunk 2 above)
Comment (storyofhis) [reply src/app/siswa/rekomendasi/_components/topsis-card.tsx]: Done!!

File: src/app/siswa/rekomendasi/_components/topsis-card.tsx (hunk 3 above)
Comment (storyofhis) [reply src/app/siswa/rekomendasi/_components/topsis-card.tsx]: Done!!

File: src/lib/api.ts (hunk 4 above)
Comment (storyofhis) [reply src/lib/api.ts]: Done!!

File: src/types/data/recommendation.ts [hunk 5]
```diff
@@ -43,4 +43,8 @@ export interface StudentRecommendationDetail {
     id: number;
     result: RecommendationResult[];
   };
+  topsis_ahp?: {
+    id: number;
+    result: RecommendationResult[];
+  };
```
Comment (albugowy15) [inline src/types/data/recommendation.ts:49]: Hapus `?` biar gak undefined

**Review by albugowy15 — CHANGES_REQUESTED** [review]

File: src/types/data/recommendation.ts (hunk 5 above)
Comment (storyofhis) [reply src/types/data/recommendation.ts]: Done!!

**Review by albugowy15 — APPROVED** [review]

**Pull request merged**